### PR TITLE
Added RunAsyncCheckException assertion

### PR DESCRIPTION
**Pull request by Pauljgraham**
This should replace RunAsync<TException> as it gives more control over which assertions are testable

### CODE REVIEW

File: SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs [hunk 1]
```diff
@@ -119,16 +119,19 @@ public async Task RunAsync<TResult>(Action<T> arrange, Func<T, Task<TResult>> ac
             assert(testFixture, actionResult);
         }
 
+        [Obsolete("Please use method RunAsycCheckException")]
```
Comment (elliotritchie) [inline SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs]: The obsolete methods can be removed and we'll bump the major version of the package.

File: SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs [hunk 2]
```diff
@@ -119,16 +119,19 @@ public async Task RunAsync<TResult>(Action<T> arrange, Func<T, Task<TResult>> ac
...
         public Task RunAsync<TException>(Func<T, Func<Task>, ExceptionAssertions<TException>> assert) where TException : Exception
         {
             return RunAsync(null, null, assert);
         }
 
+        [Obsolete("Please use method RunAsycCheckException")]
```
Comment (elliotritchie) [inline SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs]: The obsolete methods can be removed and we'll bump the major version of the package.

File: SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs [hunk 3]
```diff
@@ -119,16 +119,19 @@ public async Task RunAsync<TResult>(Action<T> arrange, Func<T, Task<TResult>> ac
...
         public Task RunAsync<TException>(Func<T, Task> act, Func<T, Func<Task>, ExceptionAssertions<TException>> assert) where TException : Exception
         {
             return RunAsync(null, act, assert);
         }
 
+        [Obsolete("Please use method RunAsycCheckException")]
```
Comment (elliotritchie) [inline SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs]: The obsolete methods can be removed and we'll bump the major version of the package.

File: SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs [hunk 4]
```diff
@@ -145,5 +148,29 @@ public async Task RunAsync<TResult>(Action<T> arrange, Func<T, Task<TResult>> ac
 
             return Task.CompletedTask;
         }
+
+        public Task RunAsyncCheckException(Func<T, Task> act,
```
Comment (elliotritchie) [inline SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs]: Could this be named `RunAsync`?

File: SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs [hunk 5]
```diff
@@ -145,5 +148,29 @@ public async Task RunAsync<TResult>(Action<T> arrange, Func<T, Task<TResult>> ac
...
+            Action<T, Func<Task>> assert)
+        {
+            return RunAsyncCheckException(null, act, assert);
+        }
+
+        public Task RunAsyncCheckException(Action<T> arrange,
```
Comment (elliotritchie) [inline SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs]: Could this be named `RunAsync`?

File: SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs (hunk 1 above)
Comment (Pauljgraham) [reply SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs]: I'd be interested in how we change version numbers for the nuget packages

File: SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs (hunk 2 above)
Comment (Pauljgraham) [reply SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs]: done

File: SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs (hunk 3 above)
Comment (Pauljgraham) [reply SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs]: dne

File: SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs (hunk 5 above)
Comment (Pauljgraham) [reply SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs]: This is where intellisense starts having problems. The checking for a thrown exception is against the SUT not the result. The intent is clearer if we explicitly know we are checking for an exception being thrown

File: SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs [hunk 6]
```diff
@@ -145,5 +148,29 @@ public async Task RunAsync<TResult>(Action<T> arrange, Func<T, Task<TResult>> ac
...
+            Func<T, Task> act,
+            Action<T, Func<Task>> assert)
+        {
+
+            if (act == null) throw new ArgumentNullException(nameof(act));
+            if (act == null) throw new ArgumentNullException(nameof(assert));
```
[the remark this answers is not in the document]
Comment (Pauljgraham) [reply SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs]: done

File: SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs (hunk 4 above)
Comment (Pauljgraham) [reply SFA.DAS.Testing/SFA.DAS.Testing/FluentTest.cs]: This is where intellisense starts having problems. The checking for a thrown exception is against the SUT not the result. The intent is clearer if we explicitly know we are checking for an exception being thrown

**Pull request merged**